mqtt_io/config/config.schema.yml

```diff
@@ -250,7 +250,24 @@ mqtt:
           type: boolean
           required: no
           default: false
-
+    reconnect_delay:
+      meta:
+        description: |
+          Time in seconds to wait between reconnect attempts.
+      type: integer
+      required: no
+      default: 2
+      min: 1
+    reconnect_count:
+      meta:
+        description: |
+          Max number of retries of connections before giving up (and exiting).
+          -1 means infinite reconnects.
+          The counter is reset if a connection was established successfully.
+      type: integer
+      required: no
+      default: 0
+      min: -1
```

Comment:
Perhaps the default should be -1, so that we try infinitely to reconnect? Presumably 0 means that we don't try to reconnect at all?

Reply:
I used 0 to be backwards compatible (Maybe someone has his/her own failure logic to handle mqtt-io failures e.g. in a parent process). But infinite reconnects are also fine.
(Yes, 0 means no reconnects)

Reply:
Actually, v0.x did support auto-reconnecting by default - it was just an oversight of mine when I rebuilt the project into v2.0 recently, so auto-reconnecting by default is what we should aim for.

Reply:
Ok, I changed the default to -1